### DISCUSSION TITLE
How to change target JS version?

### DISCUSSION THREAD

**Discussion by ZoranRavic** [vercel/next.js#69280] [Help]
By default Next.js targets an older version of JS in order to support older browsers.  
This means that it will transpile JS that uses newer features like the `?.` operator, or private class fields.

However my website uses newer browser features that can not be transpiled or polyfilled, so this transpilation is of no use to me.  
It just makes my build slightly bigger and I suspect browsers might optimize the code better if transpilation was removed.

From what I can see there is no obvious option to change the target JS version in next.config.mjs, not even thought webpack.  
And the target field from tsconfig.json is ignored.

**Comment by styfle** [comment]
> By default Next.js targets an older version of JS in order to support older browsers.

I don't think that's true. Check out the list of supported browsers here: https://nextjs.org/docs/architecture/supported-browsers

 > From what I can see there is no obvious option to change the target JS version in next.config.mjs

Right, you would need to use `browserslist` in your `package.json` file.

>  it will transpile JS that uses newer features like the ?. operator, or private class fields.

That doesn't sound right. Which version of Next.js are you using? Have you tried upgrading?

**Comment by styfle** [comment]
It sounds like your Next.js version is 4 years outdated? I found the PR where we changed the defaults:

- https://github.com/vercel/next.js/discussions/33227
